File: packages/types/child.ts
export type LearningStyle =
  | "visual" | "auditory" | "kinesthetic" | "reading_writing" | "social";

export type GradeLevel =
  | "pre_k" | "k"
  | "grade_1" | "grade_2" | "grade_3" | "grade_4" | "grade_5" | "grade_6"
  | "grade_7" | "grade_8" | "grade_9" | "grade_10" | "grade_11" | "grade_12"
  | "stage_early" | "stage_middle" | "stage_upper" | "graduated";

export interface Child {
  id: string;
  family_id: string;
  first_name: string;
  nickname?: string;
  gender?: string;
  avatar_initials?: string;
  birthdate?: string;       // ISO date string
  birth_year?: number;
  birth_month?: number;
  grade_level?: GradeLevel;
  child_curriculum: string[];
  learning_styles: LearningStyle[];
  personality_description?: string;
  personality_tags: string[];
  interests: string[];
  motivators?: string;
  demotivators?: string;
  learning_differences: string[];
  accommodations_notes?: string;
  support_services: string[];
  is_active: boolean;
  created_at: string;
}

export type ChildCreate = Omit<Child, "id" | "family_id" | "is_active" | "created_at">;

File: apps/api/app/schemas/child.py
from pydantic import BaseModel, Field
from typing import Optional
from uuid import UUID
from datetime import date, datetime


class ChildCreate(BaseModel):
    first_name: str = Field(..., min_length=1, max_length=60)
    nickname: Optional[str] = Field(None, max_length=40)
    gender: Optional[str] = None
    birthdate: Optional[date] = None
    birth_year: Optional[int] = Field(None, ge=1990, le=2030)
    birth_month: Optional[int] = Field(None, ge=1, le=12)
    grade_level: Optional[str] = None
    child_curriculum: list[str] = Field(default_factory=list)
    learning_styles: list[str] = Field(default_factory=list)
    personality_description: Optional[str] = Field(None, max_length=1000)
    personality_tags: list[str] = Field(default_factory=list)
    interests: list[str] = Field(default_factory=list)
    motivators: Optional[str] = Field(None, max_length=200)
    demotivators: Optional[str] = Field(None, max_length=200)
    learning_differences: list[str] = Field(default_factory=list)
    accommodations_notes: Optional[str] = Field(None, max_length=500)
    support_services: list[str] = Field(default_factory=list)

    model_config = {"populate_by_name": True}


class ChildUpdate(ChildCreate):
    first_name: Optional[str] = Field(None, min_length=1, max_length=60)


class ChildResponse(BaseModel):
    id: UUID
    family_id: UUID
    first_name: str
    nickname: Optional[str] = None
    gender: Optional[str] = None
    birthdate: Optional[date] = None
    birth_year: Optional[int] = None
    grade_level: Optional[str] = None
    learning_styles: list[str] = []
    personality_tags: list[str] = []
    interests: list[str] = []
    is_active: bool = True
    created_at: datetime

    model_config = {"from_attributes": True}
